src/engraving/dom/beambase.h

```diff
@@ -57,6 +91,26 @@ class BeamBase : public EngravingItem
     bool up() const { return m_up; }
     void setUp(bool v) { m_up = v; }
 
+    bool userModified() const;
+    void setUserModified(bool val);
+
+    DirectionV direction() const { return m_direction; }
+    void setDirection(DirectionV val) { m_direction = val; }
+    virtual void setBeamDirection(DirectionV v) = 0;
```

Comment:
This combination of methods looks potentially confusing. I'd rename `setDirection` to `doSetDirection` and make it protected; then rename `setBeamDirection` to `setDirection`.

Reply:
I've changed the names but `doSetDirection` needs to remain public for its use in `TremoloLayout::layoutTwoNotesTremolo`